src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/Shell.java

```diff
@@ -368,7 +369,27 @@ private void processCommandEntry()
    }
 
    public void onSendToConsole(final SendToConsoleEvent event)
-   {  
+   {
+      server_.adaptToLanguage(
```

Comment:
Does this need to be run every time we send content to the console? I'm wondering if it would be beneficial here for the Shell to know what the current language is so it doesn't have to do a server roundtrip.

Reply:
I wanted to do this just because I was afraid that this state could get out-of-sync between the session and the client some way or another. Is there a safe way to keep this all in sync?

Reply:
I suppose I could attach this state to the ConsolePromptEvent?